tests/test_client.py

```diff
@@ -33,79 +34,186 @@ def test_eval_calls_request(self, MockPath):
+                    json.dumps(expected_vars)
+                )
+
+    def test_advanced_search_user_can_view_unpublished_and_show_unpublished_is_true(self):
+        client = MarklogicApiClient("", "", "", False)
+
+        with patch.object(client, 'invoke'):
+            with patch.object(client, 'user_can_view_unpublished_judgments', return_value=True):
+                client.advanced_search(
+                    q="my-query",
+                    court="ewhc",
+                    judge="a. judge",
+                    party="a party",
+                    page=2,
+                    page_size=20,
+                    show_unpublished=True
+                )
+
+                expected_vars = {
+                    "court": "ewhc",
+                    "judge": "a. judge",
+                    "page": 2,
+                    "page-size": 20,
+                    "q": "my-query",
+                    "party": "a party",
+                    "neutral_citation": "",
+                    "specific_keyword": "",
+                    "order": "",
+                    "from": "",
+                    "to": "",
+                    "show_unpublished": "true",
+                    "only_unpublished": "false"
+                }
+
+                client.invoke.assert_called_with(
+                    '/judgments/search/search.xqy',
+                    json.dumps(expected_vars)
+                )
+
+    def test_advanced_search_user_cannot_view_unpublished_but_show_unpublished_is_true(self):
+        """The user is not permitted to see unpublished judgments but is attempting to view them
+        Set `show_unpublished` to false and log a warning"""
+        client = MarklogicApiClient("", "", "", False)
 
-    def test_eval_xslt_with_default(self):
+        with patch.object(client, 'invoke'):
+            with patch.object(client, 'user_can_view_unpublished_judgments', return_value=False):
+                with patch.object(logging, "warning") as mock_logging:
+                    client.advanced_search(
+                        q="my-query",
+                        court="ewhc",
+                        judge="a. judge",
+                        party="a party",
+                        page=2,
+                        page_size=20,
+                        show_unpublished=True
+                    )
+
+                    expected_vars = {
+                        "court": "ewhc",
+                        "judge": "a. judge",
+                        "page": 2,
+                        "page-size": 20,
+                        "q": "my-query",
+                        "party": "a party",
+                        "neutral_citation": "",
+                        "specific_keyword": "",
+                        "order": "",
+                        "from": "",
+                        "to": "",
+                        "show_unpublished": "false",
+                        "only_unpublished": "false"
```

Comment:
Does it make more sense to raise a (probably custom) error here than automatically downgrade? (I think that raising breaks things later on, perhaps?)